File: 2025/jiyeon/2025-08-10-House-Robber.java
class HouseRobber {
    public int rob(int[] nums) {
        int prevRob = 0;
        int maxRob = 0;
        for (int num : nums) {
            int temp = Math.max(maxRob, num + prevRob);
            prevRob = maxRob;
            maxRob = temp;
        }
        return maxRob;
    }
}

File: 2025/Jaehyuk/2025-08-10-House-Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        prev_rob = max_rob = 0
#so max rob is previous house which is the max val so far
#and prev rob + cur val is 2 houses ago plus current val
#[prev_rob, max_rob, cur_val, next_val, ...]
        for cur_val in nums:
            temp = max(max_rob, prev_rob + cur_val)
            prev_rob = max_rob
            max_rob = temp
        
        return max_rob